### PR TITLE
Fix Controlfreec: Add stub runs to test single sample input & make conda work with R scripts

### LINKED ISSUE #1455: [BUG] controlfreec fails with conda
Follow up to

https://github.com/nf-core/modules/pull/1452#discussion_r835546725

In the conda tests
```
cat: /usr/local/bin/assess_significance.R: No such file or directory
mv: cannot stat '*.p.value.txt': No such file or directory
```
We should figure out in assess_signifcance can just be called.

### PR DESCRIPTION
**Pull request by FriederikeHanssen**
The test data is not suited to pass as single sample through Controlfreec, because the tool requires a matched normal for wes/panel data. 

I therefore proposing to at least add a stub run test for single sample input

also closes #1455 
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/controlfreec/freec/test.yml [hunk 1]
```diff
@@ -20,4 +20,26 @@
...
+
+- name: controlfreec test_controlfreec_freec
+  command: nextflow run tests/modules/controlfreec/freec -entry test_controlfreec_freec_single -c tests/config/nextflow.config -stub-run
+  tags:
+    - controlfreec
+    - controlfreec/freec
+  files:
+    - path: output/controlfreec/config.txt
+    - path: output/controlfreec/test.mpileup.gz_control.cpn
+      md5sum: d41d8cd98f00b204e9800998ecf8427e
```
Comment (maxulysse) [inline tests/modules/controlfreec/freec/test.yml:32]: I think we don't need md5sum for empty files

Comment (FriederikeHanssen) [reply tests/modules/controlfreec/freec/test.yml:32]: ok

**Review by maxulysse — APPROVED** [review]

**Comment by FriederikeHanssen** [comment]
hm I am stuck. 

I created the controlfreec conda environment on my machine. The scripts `assess_significance.R` and `makeGrpah.R` are both available in my PATH variable. Typing their name in directly also works. As soon as I add `cat` (as mentioned in the documentation), the scripts are not available anymore. I have no idea why, googleing so far has not helped. If anyone has an idea...

**Comment by FriederikeHanssen** [comment]
@nf-core/modules-team anyone know how to get `assess_significance` to work with conda? 
I have tried setting the path to :
`~/miniconda3/envs/*/bin/` and also `~/miniconda3/envs/controlfreec/bin/`, `~/miniconda3/envs/control-freec/bin/` . I don't understand to where nextflow installs this to.

File: modules/controlfreec/assesssignificance/main.nf [hunk 2]
```diff
@@ -21,7 +21,7 @@ process CONTROLFREEC_ASSESSSIGNIFICANCE {
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
     """
-    cat /usr/local/bin/assess_significance.R | R --slave --args ${cnvs} ${ratio}
+    cat \$(which assess_significance.R) | R --slave --args ${cnvs} ${ratio}
```
Comment (maxulysse) [inline modules/controlfreec/assesssignificance/main.nf:24]: <3

**Review by JoseEspinosa — COMMENTED** [review]
LGTM!
Only one thing from the top of my head that will be indeed a more general question to discuss for all modules, adding only the `stub-run` to test the modules could not be a little bit misleading?
My point is that in fact, we are just testing several `touch` commands and that the version is correctly output but the code of the real module (i.e. the `script` section) is not tested in any way.

**Comment by FriederikeHanssen** [comment]
I agree, that is why I ahve been very hesitant to use stub tests anywhere. And also why I added both tests - once with stub and once without. The problem here is that paired test data passes, but the tools also accepts unpaired data. Unfortuantely, the test data we have makes the tool then complain ( would need full size wgs data as far as I know). 

Also I wanted to add some CI tests to sarek, which then could also only test the paired subworkflow but not the unpaired one. While not ideal to just test the file existence it now at least allows me to test that the subworkflow as a whole works without modifying the module code locally.

**Comment by FriederikeHanssen** [comment]
Conda test now fails because of mismatching md5sum. Ignoring this

**Comment by JoseEspinosa** [comment]
OK, fair enough :rocket:

**Pull request merged**